### PR TITLE
test: fix unstable TestMemoryLeak

### LINKED ISSUE #28049: Unstable test `session_test.go:5663: testTiDBAsLibrary.TestMemoryLeak`
## Bug Report

Please answer these questions before submitting your issue. Thanks!
```
[2021-09-14T21:36:47.777Z] ----------------------------------------------------------------------
[2021-09-14T21:36:47.777Z] FAIL: session_test.go:5663: testTiDBAsLibrary.TestMemoryLeak
[2021-09-14T21:36:47.777Z] 
[2021-09-14T21:36:47.777Z] session_test.go:5686:
[2021-09-14T21:36:47.777Z]     c.Assert(memStat.HeapInuse-oldHeapInUse, Less, uint64(150*units.MiB))
[2021-09-14T21:36:47.777Z] ... compare_one uint64 = 0xae3a000
[2021-09-14T21:36:47.777Z] ... compare_two uint64 = 0x9600000
[2021-09-14T21:36:47.777Z] 
[2021-09-14T21:36:47.777Z] OOPS: 263 passed, 3 skipped, 1 FAILED
[2021-09-14T21:36:47.777Z] --- FAIL: TestT (191.48s)
[2021-09-14T21:36:47.777Z] FAIL
[2021-09-14T21:36:48.344Z] FAIL	github.com/pingcap/tidb/session	192.379s
[2021-09-14T21:36:48.344Z] FAIL
```

### 1. Minimal reproduce step (Required)
in ci https://ci.pingcap.net/blue/rest/organizations/jenkins/pipelines/tidb-unit-test-nightly/runs/10732/nodes/113/steps/853/log/?start=0

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #28049

Problem Summary:

### What is changed and how it works?

What's Changed:

increase the repeat times and memory limit, to tolerate unstable GC(?)

How it Works:

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by tisonkun** [comment]
/cc @morgo @xhebox 

Could you please take a look.

**Review by xhebox — APPROVED** [review]

File: session/session_test.go [hunk 1]
```diff
@@ -5677,13 +5677,14 @@ func (s *testTiDBAsLibrary) TestMemoryLeak(c *C) {
...
-	for i := 0; i < 10; i++ {
+	for i := 0; i < 20; i++ {
 		initAndCloseTiDB()
 	}
 
 	runtime.GC()
 	runtime.ReadMemStats(&memStat)
-	c.Assert(memStat.HeapInuse-oldHeapInUse, Less, uint64(150*units.MiB))
+	// before the fix, initAndCloseTiDB for 20 times will cost 900 MB memory, so we test for a quite loose upper bound.
+	c.Assert(memStat.HeapInuse-oldHeapInUse, Less, uint64(300*units.MiB))
```
Comment (morgo) [inline session/session_test.go:5687]: Can you explain how this is expected to work?

I would have thought either `oldHeapInUse` is calculated after initAndCloseTiDB() has been executed 20 times.

Comment (lance6716) [reply session/session_test.go:5687]: this unstable test is introduced by https://github.com/pingcap/tidb/pull/28027 , after dom.Close some memory didn't get released.

so repeat some times, leak grows significantly than usual memory. more repeat times gives us a wider bound to assert

**Comment by lance6716** [comment]
ping @morgo @wjhuang2016

**Comment by zhouqiang-cl** [comment]
ping @lance6716

**Comment by lance6716** [comment]
ping @morgo @wjhuang2016

**Review by morgo — APPROVED** [review]
LGTM

**Pull request merged**